javascript/ql/src/experimental/Security/CWE-614/InsecureCookie.qll

```diff
@@ -0,0 +1,161 @@
+/**
+ * Provides classes for reasoning about cookies added to response without the 'secure' flag being set.
+ * A cookie without the 'secure' flag being set can be intercepted and read by a malicious user.
+ */
+
+import javascript
+
+module InsecureCookie {
+  /**
+   * `secure` property of the cookie options.
+   */
+  string flag() { result = "secure" }
+
+  /**
+   * Abstract class to represent different cases of insecure cookie settings.
+   */
+  abstract class InsecureCookies extends DataFlow::Node {
+    /**
+     * The name of the middleware/library used to set the cookie.
+     */
+    abstract string getKind();
+
+    /**
+     * The `cookie` options.
+     */
+    abstract DataFlow::Node getCookieOptionsArgument();
+
+    /**
+     * Predicate that determines if a cookie is insecure.
+     */
+    abstract predicate isInsecure();
+  }
+
+  /**
+   * A cookie set using the `express` module `cookie-session` (https://github.com/expressjs/cookie-session).
+   * The flag `secure` is set to `false` by default for HTTP, `true` by default for HTTPS (https://github.com/expressjs/cookie-session#cookie-options).
+   */
+  class InsecureCookieSession extends ExpressLibraries::CookieSession::MiddlewareInstance,
+    InsecureCookies {
+    InsecureCookieSession() { this instanceof ExpressLibraries::CookieSession::MiddlewareInstance }
+
+    override string getKind() { result = "cookie-session" }
+
+    override DataFlow::SourceNode getCookieOptionsArgument() {
+      result = this.getOption("cookie").(DataFlow::SourceNode)
```

Comment:
```suggestion
      result = this.getOption("cookie")
```
This cast is redundant, the `result`-type in the signature of this predicate performs the same cast.